addons/base_import/static/src/additionalFileManager.js

```diff
@@ -0,0 +1,93 @@
+import { checkFileSize } from "@web/core/utils/files";
+
+export class AdditionalFileManager {
+    constructor(resModel, fields, context, options, orm, notificationService, maxBatchSize) {
+        this.resModel = resModel;
+        this.fields = [".id", ...fields];
+
+        this.context = context;
+        this.options = options;
+        this.orm = orm;
+        this.notificationService = notificationService;
+        this.maxBatchSize = maxBatchSize * 0.95; // 0.95 for not calculated payload overhead
+
+        this.dataToSend = {};
+        this.lastSend = Date.now();
+        this.delayAfterEachBatch = this.options.delayAfterEachBatch * 1000;
+    }
+
+    async addFile(id, field, file) {
+        let data = await this._readFile(file);
+        if (typeof data === "string" && data.startsWith("data:")) {
+            // Remove data:image/*;base64,
+            data = data.split(",")[1];
+        }
+        const dataSize = data.length;
+        if (!checkFileSize(dataSize, this.notificationService)) {
+            return;
+        }
+
+        if (this.getCurrentSize() + dataSize >= this.maxBatchSize) {
+            await this._send();
+        }
+        if (!(id in this.dataToSend)) {
+            this.dataToSend[id] = Array(this.fields.length);
+            this.dataToSend[id][0] = id;
+        }
+        const indexOfField = this.fields.indexOf(field, 1);
+        this.dataToSend[id][indexOfField] = data;
+    }
+
+    async sendLastPayload() {
+        if (Object.keys(this.dataToSend).length > 0) {
+            await this._send();
+        }
+    }
+
+    async _send() {
+        const now = Date.now();
+        if (this.lastSend + this.delayAfterEachBatch <= now) {
```

Comment:
lastSend never updated? Moreover, I think the condition is wrong (`<= now` means that it was a long time ago, so why do we wait in that case?)

Reply:
Also, why do we need this protection?

Reply:
Old logic now we always wait for `delayAfterEachBatch` before another call. I forgot to remove it.